crates/consensus/common/src/validation.rs

```diff
@@ -289,18 +294,32 @@ pub fn validate_header_regarding_parent(
     }
 
     // Check gas limit, max diff between child/parent gas_limit should be  max_diff=parent_gas/1024
-    if child.gas_limit > parent_gas_limit {
-        if child.gas_limit - parent_gas_limit >= parent_gas_limit / 1024 {
-            return Err(ConsensusError::GasLimitInvalidIncrease {
+    // On Optimism, the gas limit can adjust instantly, so we skip this check if the optimism
+    // feature flag is enabled.
+    // TODO(clabby): Should probably do this at runtime by checking the chainspec.
+    #[cfg(not(feature = "optimism"))]
+    {
+        let mut parent_gas_limit = parent.gas_limit;
+
+        // By consensus, gas_limit is multiplied by elasticity (*2) on
+        // on exact block that hardfork happens.
+        if chain_spec.fork(Hardfork::London).transitions_at_block(child.number) {
+            parent_gas_limit = parent.gas_limit * constants::EIP1559_ELASTICITY_MULTIPLIER;
+        }
```

Comment:
It looks like we have these block scope variables defined as the same values in the parent scope so we shouldn't need to "mask" these variables and have the parent scope variables be "shadowed".